components/validation/src/main/java/org/wso2/carbon/identity/x509Certificate/validation/CertificateValidationUtil.java

```diff
@@ -1494,4 +1486,551 @@ private static Validator resourceToValidatorObject(
+            String serialNumber = getNormalizedName(certificate.getSerialNumber().toString());
+
+            List<String> ocspUrls = getValidationUrls(certificate, tenantDomain, OCSP_VALIDATOR);
+            List<String> crlUrls = getValidationUrls(certificate, tenantDomain, CRL_VALIDATOR);
+
+            Certificate newCert = new Certificate.Builder()
+                    .name(X509_CERT_PREFIX + UUID.randomUUID())
+                    .certificateContent(encodeCertificate(certificate))
+                    .build();
+            String certId = CertValidationDataHolder.getInstance().getCertificateManagementService()
+                    .addCertificate(newCert, tenantDomain);
+
+            CertObject certObject = new CertObject();
+            certObject.setCertId(certId);
+            certObject.setSerialNumber(serialNumber);
+            certObject.setCrlUrls(crlUrls);
+            certObject.setOcspUrls(ocspUrls);
+
+            if (issuerDNMap.getIssuerCertMap().containsKey(issuerDN)) {
+                List<CertObject> certList = issuerDNMap.getIssuerCertMap().get(issuerDN);
+
+                boolean serialExists = certList.stream()
+                        .anyMatch(cert -> getNormalizedName(cert.getSerialNumber()).equals(serialNumber));
+
+                if (serialExists) {
+                    throw new CertificateValidationException("Certificate with the serial number: " + serialNumber +
+                            " already exists for the issuerDN: " + issuerDN);
+                } else {
+                    log.debug("Adding new certificate with serial number: " + serialNumber + " for issuerDN: " +
+                            issuerDN);
+                    certList.add(certObject);
+                    issuerDNMap.getIssuerCertMap().put(issuerDN, certList);
+                }
+            } else {
+                // Add new issuerDN with the cert
+                List<CertObject> newCertList = new ArrayList<>();
+                newCertList.add(certObject);
+                issuerDNMap.getIssuerCertMap().put(issuerDN, newCertList);
+            }
+            saveUpdatedResource(resource, issuerDNMap);
+            return Optional.of(certObject);
+        } catch (ConfigurationManagementException | IOException | CertificateException | CertificateMgtException e) {
+            throw new CertificateValidationException("Error while processing the resource", e);
+        }
+    }
+
+    /**
+     * Update a certificate in the configuration store by certificate id.
+     *
+     * @param certificateId Certificate Id
+     * @param certificate   X509Certificate
+     * @param tenantDomain  Tenant Domain
+     * @return CertObject
+     * @throws CertificateValidationException Error when updating certificate
+     * @throws CertificateException           Error when encoding certificate
+     * @throws CertificateMgtException        Error when updating certificate
```

Comment:
Let's add fullstops at the end of comment lines. Let's check other places as well.